xapian-core/tests/harness/testsuite.cc

```diff
@@ -114,8 +114,11 @@ string test_driver::col_red, test_driver::col_green;
 string test_driver::col_yellow, test_driver::col_reset;
 bool test_driver::use_cr = false;
 
-// time constant to mark tests as slow or not
-static const int SLOW_TEST_THRESHOLD = 10000;
+// time constant in seconds to mark tests as slow or not
+static const float SLOW_TEST_THRESHOLD = 10.00;
```

Comment:
`double` is a better default choice of floating point type than `float`.  The exceptions that come to mind are if you're storing so many values that space becomes a concern (not the case here), or you don't have hardware floating point in which case `float` might be faster.  Otherwise `double` offer a much better precision for a likely very similar speed.

There are in fact currently no uses of the `float` type anywhere in the xapian source tree:

```
olly@gemse:~/git/xapian$ git grep -w float|grep '\.\(cc\|h\):'
xapian-core/ChangeLog:	* api/matchspy.cc: Use <cfloat> and <cmath> rather than <float.h> and
xapian-core/api/queryinternal.cc:	// results to float (which is another approach which works).
xapian-core/api/sortable-serialise.cc:	// INFINITY is C99.  Oddly, it's of type "float" so sanity check in
```

Reply:
changed float to double.